Lib/http/cookiejar.py

```diff
@@ -532,15 +532,31 @@ def parse_ns_headers(ns_headers):
     return result
 
 
+# only kept for backwards compatibilty.
 IPV4_RE = re.compile(r"\.\d+$", re.ASCII)
+
+def is_ip_like_hostname(text: str):
+    """Return True if text is a valid hostname in the form of IP address."""
+    from ipaddress import IPv4Address, IPv6Address
+    # check for IPv4 address
+    try:
+        IPv4Address(text)
+    except ValueError:
+        # check for IPv6 address in []
+        if text.startswith('[') and text.endswith(']'):
+            try:
+                IPv6Address(text.removeprefix('[').removesuffix(']'))
+            except ValueError:
+                return False
```

Comment:
```suggestion
        if text.startswith('[') and text.endswith(']'):
            try:
                IPv6Address(text[1:-1])
            except ValueError:
                return False
```

Now we can do this as we're checking for startswith and endswith.